compiler/z/codegen/S390Peephole.cpp

```diff
@@ -1349,29 +1349,89 @@ TR_S390PostRAPeephole::ConditionalBranchReduction(TR::InstOpCode::Mnemonic branc
 
 
 /**
- * Catch the pattern where an CR/BRC can be converted
- *    CR R1,R2
- *    BRC  Mask, Lable
+ * Catch the pattern where an CLR/BRC can be converted
+ *    CLR R1, R2
+ *    BRC Mask, Lable
  * Can be replaced with
- *    CRJ  R1,R2,Lable
+ *    CLRJ R1, R2, Lable, Mask
  */
 bool
 TR_S390PostRAPeephole::CompareAndBranchReduction()
    {
-   bool performed = false;
+   if (!TR::Compiler->target.cpu.getSupportsArch(TR::CPU::z10))
+      return false;
 
-   TR::Instruction *current = _cursor;
-   TR::Instruction *next = _cursor->getNext();
-   TR::InstOpCode::Mnemonic curOpCode = current->getOpCodeValue();
-   TR::InstOpCode::Mnemonic nextOpCode = next->getOpCodeValue();
-   if ((curOpCode == TR::InstOpCode::CR || curOpCode == TR::InstOpCode::CGR || curOpCode == TR::InstOpCode::CGFR)
-      && (nextOpCode == TR::InstOpCode::BRC || nextOpCode == TR::InstOpCode::BRCL))
-      {
-      printf("Finding CR + BRC\n");
-      printf("method=%s\n", comp()->signature());
+   bool branchTakenPerformReduction = false;
+   bool fallThroughPerformReduction = false;
+   TR::Instruction *clrInstruction = _cursor;
+   TR::Instruction *brcInstruction = _cursor->getNext();
+   TR::Instruction *clrjInstruction = NULL;
+   TR::LabelSymbol *labelSymbol = NULL;
```

Comment:
Some of these variables are only used within the `if` block. We can limit their scope by declaring them at the latest possible time. This also improves code readability as it limits the lifetime of variables (think debugging and reading code).

The variables `clrInstruction` and `brcInstruction` are a little misleading to me, as their values are only valid within the `if` block below. If a reader is not careful to examine the whole function one may assume that `brcInstruction` variable will hold a pointer to a `BRC` even outside the `if` block which is simply not true. Limiting the scope of variables to only blocks in which they are valid and assigned prevents such issues.